core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -253,6 +253,21 @@ ClientRequestContext newDerivedContext(@Nullable HttpRequest req, @Nullable RpcR
      */
     void setResponseTimeout(Duration responseTimeout);
 
+    /**
+     * Returns {@link Response} timeout handler which is executed when
+     * the {@link Response} is not completely received within the allowed {@link #responseTimeoutMillis()}
+     * or the default {@link ClientOption#RESPONSE_TIMEOUT_MILLIS}.
+     */
+    @Nullable
+    Runnable responseTimeoutHandler();
+
+    /**
```

Comment:
Think we should document something like this:

```java
/**
     * Sets a handler to run when the response times out. {@code responseTimeoutHandler} must close the request,
     * e.g., by calling {@link HttpRequestWriter#close(Throwable)}. If not set, the response will be closed with
     * {@link TimeoutException}.
     *
     * <p>For example,
     * <pre>{@code
     *   HttpRequestWriter req = HttpRequest.streaming();
     *   ctx.setResponseTimeoutHandler(() -> {
     *      req.close(new IllegalStateException("Server is in a bad state."));
     *   });
     *   ...
     * }</pre>
     */
```

Reply:
Thanks for the kind review! I didn't know you have been working on this issue... 😱 I should've shared working on that 😭

Reply:
No problem, I didn't share either nor get back to the branch. Glad you send a PR :)